src/libs/PaginationUtils.ts

```diff
@@ -0,0 +1,191 @@
+import CONST from '@src/CONST';
+import type Pages from '@src/types/onyx/Pages';
+
+type PageWithIndex = {
+    /** The IDs we store in Onyx and which make up the page. */
+    ids: string[];
+
+    /** The first ID in the page. */
+    firstID: string;
+
+    /** The index of the first ID in the page in the complete set of sorted items. */
+    firstIndex: number;
+
+    /** The last ID in the page. */
+    lastID: string;
+
+    /** The index of the last ID in the page in the complete set of sorted items. */
+    lastIndex: number;
+};
+
+// It's useful to be able to reference and item along with its index in a sorted array,
+// since the index is needed for ordering but the id is what we actually store.
+type ItemWithIndex = {
+    id: string;
+    index: number;
+};
+
+/**
+ * Finds the id and index in sortedItems of the first item in the given page that's present in sortedItems.
+ */
+function findFirstItem<TResource>(sortedItems: TResource[], page: string[], getID: (item: TResource) => string): ItemWithIndex | null {
+    for (const id of page) {
+        if (id === CONST.PAGINATION_START_ID) {
+            return {id, index: 0};
+        }
+        const index = sortedItems.findIndex((item) => getID(item) === id);
+        if (index !== -1) {
+            return {id, index};
+        }
+    }
+    return null;
+}
+
+/**
+ * Finds the id and index in sortedItems of the last item in the given page that's present in sortedItems.
+ */
+function findLastItem<TResource>(sortedItems: TResource[], page: string[], getID: (item: TResource) => string): ItemWithIndex | null {
+    for (const id of page.slice().reverse()) {
```

Comment:
Isn't better to just do a inverted for-loop instead of copying and reverting the whole array?

Reply:
updated